packages/trimerge-sync-basic-server/src/lib/SqliteDocStore.ts

```diff
@@ -60,40 +66,55 @@ export class SqliteDocStore implements DocStore {
     const stmt =
       lastSyncId === undefined
         ? this.db.prepare(
-            `SELECT * FROM commits ORDER BY remoteSyncId, remoteSyncIndex`,
-          )
+          `SELECT * FROM commits ORDER BY remoteSyncId, remoteSyncIndex`,
+        )
         : this.db.prepare(
-            `SELECT * FROM commits WHERE remoteSyncId > @lastSyncId ORDER BY remoteSyncId, remoteSyncIndex`,
-          );
+          `SELECT * FROM commits WHERE remoteSyncId > @lastSyncId ORDER BY remoteSyncId, remoteSyncIndex`,
+        );
 
     const sqliteCommits: SqliteCommitType[] = stmt.all({ lastSyncId });
     let syncId = '';
     const commits = sqliteCommits.map(
-      ({
-        ref,
-        remoteSyncId,
-        userId,
-        clientId,
-        baseRef,
-        mergeRef,
-        mergeBaseRef,
-        delta,
-        editMetadata,
-      }): Commit<unknown, unknown> => {
-        if (remoteSyncId) {
-          syncId = remoteSyncId;
-        }
-        return {
+      (commit): ServerCommit<unknown, unknown> => {
+        const {
           ref,
           remoteSyncId,
           userId,
-          clientId,
-          baseRef: baseRef || undefined,
-          mergeRef: mergeRef || undefined,
-          mergeBaseRef: mergeBaseRef || undefined,
-          delta: delta ? JSON.parse(delta) : undefined,
-          editMetadata: editMetadata ? JSON.parse(editMetadata) : undefined,
-        };
+          baseRef,
+          mergeRef,
+          mergeBaseRef,
+          delta,
+          metadata,
+          main,
+        } = commit;
+
+        if (remoteSyncId) {
+          syncId = remoteSyncId;
+        }
+
+        if (mergeRef) {
+          return {
+            ref,
+            remoteSyncId,
+            userId,
+            baseRef: baseRef || undefined,
+            mergeRef: mergeRef || undefined,
+            mergeBaseRef: mergeBaseRef || undefined,
+            delta: delta ? JSON.parse(delta) : undefined,
+            metadata: metadata ? JSON.parse(metadata) : undefined,
+            main,
+          } as ServerCommit<unknown, unknown>;
```

Comment:
is this cast needed?

Reply:
okay yeah I just misread the type errors. Will restructure this.